### PR TITLE
DPL 997 - PacBio support libraries in run creation

### LINKED ISSUE #1423: DPL-997 - As developers we need to modify the run creation process so that it allows pools and libraries to be added to a well as this matches the lab process better and supports Revio multiplexing
**User story**
As developers we need to modify the run creation process so that it allows pools and libraries to be added to a well as this matches the lab process better and supports Revio multiplexing

**Who are the primary contacts for this story**
James / Steve

**Who is the nominated tester for UAT**
PSD

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] pools and libraries can be searched for in the PacBio run show page
- [ ] Libraries can be added to wells in the PacBio run show page
- [ ] When loading an existing run it should pull back both pools and libraries
- [ ] update Resolve Operations to improve performance https://vitejs.dev/guide/performance.html

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

References
This story has a non-blocking relationship with:
- #sanger/traction-service#1137
- sanger/<repo>#<issue_no.>

**Additional context**

### PR DESCRIPTION
**Pull request by BenTopping**
Closes #1423 

This PR is in conjunction with: https://github.com/sanger/traction-service/pull/1218

#### Changes proposed in this pull request

- Adds PacbioLibraryTubeItem to represent library data on the pacbio run show page.
- Updates pacbioRunCreate store to allow both libraries and pools to be used when creating a run.

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: src/stores/pacbioRunCreate.js [hunk 1]
```diff
@@ -90,53 +133,47 @@ export const usePacbioRunCreateStore = defineStore('pacbioRunCreate', {
...
+    /**
+     * Returns a list of all the tubes with their contents (pool or library)
+     * @param {Object} state the pinia state object
+     * @returns
+     */
+    tubeContents: (state) => {
+      return Object.values(state.tubes).reduce((result, tube) => {
+        // We should assume a tube only has one pool
+        const pool = state.pools[tube.pools?.[0]]
+        const library = state.libraries[tube.libraries]
```
Comment (seenanair) [inline src/stores/pacbioRunCreate.js:160]: Just a doubt- what does 'tube.libraries' return? I hope it is not an array (as for pools)?

File: src/stores/pacbioRunCreate.js [hunk 2]
```diff
@@ -249,12 +283,10 @@ export const usePacbioRunCreateStore = defineStore('pacbioRunCreate', {
         id,
         // This is long but we want to include pool data
```
Comment (seenanair) [inline src/stores/pacbioRunCreate.js:284]: May be good to update the documentation as we are including library data as well

File: src/stores/pacbioRunCreate.js (hunk 1 above)
Comment (BenTopping) [reply src/stores/pacbioRunCreate.js:160]: No its just a singular library. The naming is a bit annoying but its to make it consistent with the service. Libraries is a singular id.

File: src/components/pacbio/PacbioPoolTubeItem.vue [hunk 3]
```diff
@@ -2,169 +2,147 @@
...
+              <dd>
+                <ul>
+                  <li v-for="(error, index) in errors" :key="index">
+                    {{ error }}
+                  </li>
+                </ul>
+              </dd>
+            </dl>
+          </div>
+          <div v-else>Pool invalid. Click for more information</div>
```
[the remark this answers is not in the document]
Comment (BenTopping) [reply src/components/pacbio/PacbioPoolTubeItem.vue:76]: Yeah good point, lots of shared behaviour and design. I will give it a go.

[the remark this answers is not in the document]
Comment (BenTopping) [reply src/components/pacbio/PacbioPoolTubeItem.vue]: I have update it 👍

**Review by seenanair — APPROVED** [review]
Cool, All looking good

**Pull request merged**